src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -187,6 +188,7 @@ export class MockHttpSocket extends MockSocket {
       .on('prefinish', () => this.emit('prefinish'))
       .on('finish', () => this.emit('finish'))
       .on('close', (hadError) => this.emit('close', hadError))
+      .on('end', () => this.emit('end'))
```

Comment:
We never forwarded the `end` event of the original socket. I think this should be added.